### DISCUSSION TITLE
Apply discount to subtotal of cart

### DISCUSSION THREAD

**Discussion by mirko-pira** [medusajs/medusa#3342] [Q&A]
A simple question, is it possible at this time to technically apply the discount to the subtotal of the cart (price without tax) and NOT to the total of the cart (including tax)?

**Comment by srindom** [comment]
Hi @mirko-pira, discounts are always calculated off of the subtotal.

E.g. let's say you have two items in your cart subject to a 10% tax rate and you want to apply a 30% discount, you'd get the following:

- Item A - $10 ex tax
  - Discount adjustment: $3
- Item B - $20 ex tax
  - Discount adjustment: $6

The cart totals would come out to:
- Subtotal: $30
- Discount: $9
- Taxable amount: $21
- Tax: $2.10
- Total: $23.10

Let me know if this is helpful or if I misunderstood your question? :)

**Reply by pepijn-vanvlaanderen** [reply]
@srindom this calculation seems suitable when pricing is based on excluding tax. However when having a store where all prices are including tax, calculations should behave differently in my opnion.

An example:
- Item A - $10 incl tax
  - Discount adjustment: $3
  - Expected total incl tax for the item $7
- Item B - $20 incl tax
  - Discount adjustment: $6
  - Expected total incl tax for the item $14

Currently the cart totals would come out to:
- Subtotal: $30
- Discount: $9
- Taxable amount: $21
- Tax: $2.10
- Total: $23.10 
  - While from the example incl tax you would expect a total of $21

Therefore when considering prices are including tax, one would expect the discount to be applied like:
- Subtotal: $30 (incl tax)
- Discount: $9
- Total: $21
- Taxable amount: $21 / (100 + TAX_RATE) * 100 = $19.09
- Tax: $21 / (100 + TAX_RATE) * TAX_RATE = $1.91

So to make this discussion a bit more broader, I think the discounts in combination with includes_tax should be altered in the behavior. Is this something that you want to investigate further @srindom or want some help on?

Another thing is that currently not working when includes tax is used, are fixed amount coupons (on total amount). This results in a server error, I will create a PR with a fix for that.
